decimal/fuzzing_test.go

```diff
@@ -0,0 +1,46 @@
+//go:build fuzzing
+// +build fuzzing
+
+package decimal_test
+
+import (
+	"testing"
+
+	"github.com/shopspring/decimal"
+	. "github.com/tarantool/go-tarantool/decimal"
+)
+
+func strToDecimal(t *testing.T, buf string) decimal.Decimal {
+	decNum, err := decimal.NewFromString(buf)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return decNum
+}
+
+func FuzzEncodeDecodeBCD(f *testing.F) {
+	samples := append(correctnessSamples, benchmarkSamples...)
+	for _, testcase := range samples {
+		if len(testcase.numString) > 0 {
+			f.Add(testcase.numString) // Use f.Add to provide a seed corpus.
+		}
+	}
+	f.Fuzz(func(t *testing.T, orig string) {
+		if l := GetNumberLength(orig); l > DecimalPrecision {
+			t.Skip("max number length is exceeded")
+		}
+		bcdBuf, err := EncodeStringToBCD(orig)
+		if err != nil {
```

Comment:
So what should happen if we try to insert invalid input? msgpack encoding will throw an error?

Reply:
I don't see MsgPack encoding here. And you?
Regarding BCD encoding - yes.